### PR TITLE
GAWB-2532 related v2: policy names singular, return emails in listGroups endpoint

### PR DESCRIPTION
**Pull request by MatthewBemis**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala [hunk 1]
```diff
@@ -87,8 +87,18 @@ class ManagedGroupService(private val resourceService: ResourceService, private
...
+      val emailsToLookup = ripns.map(x => WorkbenchGroupName(x.resourceId.value))
+
+      directoryDAO.batchLoadGroupEmail(emailsToLookup).map { emailLookup =>
+        val emailLookupMap = emailLookup.toMap
+
+        ripns.map { ripn =>
+          ManagedGroupMembershipEntry(ripn.resourceId, ripn.accessPolicyName, emailLookupMap(WorkbenchGroupName(ripn.resourceId.value)))
+        }
+      }
+    }
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala]: this would probably look cleaner as a for comprehension

**Review by dvoet — APPROVED** [review]
1 style comment, take it or leave it.

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by MatthewBemis** [comment]
fiab-stop failed because an unrelated script problem, but all of the tests passed. automation run still declared a success by QA and devops

**Pull request merged**